.github/workflows/ci-pipeline.yaml

```diff
@@ -0,0 +1,35 @@
+name: Continuous Integration
+on:
+  push:
+    paths-ignore:
+      - "README.md"
+  pull_request:
+    paths-ignore:
+      - "README.md"
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: docker build --tag todo-app:test --target test --platform linux/amd64 .
+      - run: docker run todo-app:test
+  deploy:
+    name: Publish to Docker Hub and deploy to Azure
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Login to Docker
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build image
+        run: docker build --tag ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:prod --target production --platform linux/amd64 .
+      - name: Publish to docker hub
+        run: docker push ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:prod
+      - name: Trigger deployment to azure
+        run: curl -dH -X POST '${{ secrets.AZURE_APP_WEBHOOK }}'
```

Comment:
Not important, but there's an interesting outcome in the logs here; your curl response shows in one line:
> {"OperationId":"399da923-52d2-40e4-ae85-ef6ad785b408","TrackingUrl":"https://***-todo-app-2024.scm.azurewebsites.net/api/logstream?filter=op:399da923-52d2-40e4-ae85-ef6ad785b408,volatile:false"}

And if we look carefully at the tracking URL, we can make a pretty strong guess that the `***` is probably redacted because of your DOCKERHUB_USERNAME secret. Working backwards from the website URL - it's pretty clear that the username is `thebau`! Now in this case the username could as easily be a variable as a secret, it's not particularly sensitive, so no harm done, but if we had a poor quality password (like some key words related to the project, e.g. `todo-app` or something) this redaction can actually _reveal_ sensitive information ironically!

Nothing I'd actually change, but worth understanding a weird potential risk